src/electron/build.action.mjs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {getBuildParameters} from '../build/get_build_parameters.mjs';
+import {getVersion} from '../build/get_version.mjs';
+import {runAction} from '../build/run_action.mjs';
+import electron, {Platform} from 'electron-builder';
+import fs from 'fs/promises';
+import url from 'url';
+import {getRootDir} from '../build/get_root_dir.mjs';
+import path from 'path';
+
+export async function main(...parameters) {
+  const {platform, buildMode, stagingPercentage} = getBuildParameters(parameters);
+  const version = await getVersion(platform);
+
+  await runAction('www/build', platform, `--buildMode=${buildMode}`);
+  await runAction('electron/build_main', ...parameters);
+  await runAction('electron/build_icons', '--input=resources/electron/icon.png', '--output=build');
```

Comment:
What's the output flag? Shouldn't you use the output dir env var?

Reply:
These are just inputs to `electron-build-icons` - i can bake them directly into the action if that's less confusing, i thought i might just expose the flexibility, but it's not necessary